assembly/pom.xml

```diff
@@ -166,6 +166,12 @@
           <version>${project.version}</version>
           <scope>provided</scope>
         </dependency>
+        <dependency>
```

Comment:
This is needed to avoid `java.lang.NoClassDefFoundError: org/apache/spark/sql/protobuf/CatalystDataToProtobuf`. Similar issue was also mentioned in Avro support [PR](https://github.com/apache/spark/pull/40535/files#r1150299249).